lib/web_ui/dev/integration_tests_manager.dart

```diff
@@ -121,47 +125,87 @@ class IntegrationTestsManager {
     final List<String> blockedTests =
         blockedTestsListsMap[getBlockedTestsListMapKey(_browser)] ?? <String>[];
 
-    // The following loops over the contents of the directory and saves an
-    // expected driver file name for each e2e test assuming any dart file
-    // not ending with `_test.dart` is an e2e test.
-    // Other files are not considered since developers can add files such as
-    // README.
-    for (io.File f in entities) {
-      final String basename = pathlib.basename(f.path);
-      if (!basename.contains('_test.dart') && basename.endsWith('.dart')) {
-        // Do not add the basename if it is in the `blockedTests`.
-        if (!blockedTests.contains(basename)) {
-          e2eTestsToRun.add(basename);
-        } else {
-          print('INFO: Test $basename is skipped since it is blocked for '
-              '${getBlockedTestsListMapKey(_browser)}');
+    // If no target is specified run all the tests.
+    if (_runAllTestTargets) {
+      // The following loops over the contents of the directory and saves an
+      // expected driver file name for each e2e test assuming any dart file
+      // not ending with `_test.dart` is an e2e test.
+      // Other files are not considered since developers can add files such as
+      // README.
+      for (io.File f in entities) {
+        final String basename = pathlib.basename(f.path);
+        if (!basename.contains('_test.dart') && basename.endsWith('.dart')) {
+          // Do not add the basename if it is in the `blockedTests`.
+          if (!blockedTests.contains(basename)) {
+            e2eTestsToRun.add(basename);
+          } else {
+            print('INFO: Test $basename is skipped since it is blocked for '
+                '${getBlockedTestsListMapKey(_browser)}');
+          }
         }
       }
+      print(
+          'INFO: In project ${directory} ${e2eTestsToRun.length} tests to run.');
+    } else {
+      // If a target is specified it will run regardless of if it's blocked or
+      // not. There will be an info note to warn the developer.
+      final String targetTest =
+          IntegrationTestsArgumentParser.instance.testTarget;
+      final io.File file =
+          entities.singleWhere((f) => pathlib.basename(f.path) == targetTest);
+      final String basename = pathlib.basename(file.path);
+      if (blockedTests.contains(basename)) {
+        print('INFO: Test $basename do not run on CI environments.');
+      }
+      e2eTestsToRun.add(basename);
```

Comment:
Should this be inside an `else` case? We shouldn't add it if it's blocked, right?

Reply:
No, this path only works if the developer explicitly gives the test target. For example even if a test is in the blocked list, it will run when the developer asks for that target. I thought it would be a useful option to provide for developers.

The warning is saying, this test is not run on CI environments to inform the developer.

Reply:
Ah, I see. `felt test` can be quite noisy, especially when it clones the Flutter repo and installs drivers, which makes it hard to notices messages like that. This can mislead one into thinking that they fixed the test and that it now passes on CI while in fact they need to edit some felt code to enable it. If a test is blocked I'd prefer that felt fail with a message explaining how to unblock the test before running it.

Reply:
I added the message after `verbose`.

`felt test` won't run this test, it will only run if one runs: `felt test --integration-tests-only target_that_is_skipped` The reason I'm keeping it that it is easier to use felt (downloads correct version flutter/flutter, installs driver). If someone has a test they want to fix, it will be easier to work on it this way.